### PR TITLE
Add base64

### PR DESCRIPTION
**Pull request by YoungSeokHong**
I made Base64 in the same form as Base32. And also made base64Test for gtest. Please make some test case in discussion #87 
Thank you!

### CODE REVIEW

**Review by yeonns — COMMENTED** [review]
It also includes base32 feature (duplicated PR #79) 😮

**Comment by YoungSeokHong** [comment]
> It also includes base32 feature (duplicated PR #79) 😮

Thank you! I think it's because that I made base64 branch from base32 branch. I didn't realize it became duplicated.
Thanks for your review!

**Comment by yeonns** [comment]
@YoungSeokHong Sorry for making you remove the duplicate code. 😢 
Maybe the process below is simpler. I think you can try it next time. 😄 
```sh
git checkout -b base64
git reset --hard origin master
git cherry-pick 909b109 # cherry pick Base64 related commit
git cherry-pick f827bcb # cherry pick Base64 related commit
git cherry-pick ...     # if you have more Base64 related commit, you can cherry-pick
git push -f origin base64:added_base64 # force push from local base64 branch to origin added_base64 branch
```

**Comment by aiclaudev** [comment]
Thanks for your hard work!!

**Comment by Jsueeee** [comment]
Great job👍👍👍👍 Thanks for your contribution!

**Comment by ycs1m1yk** [comment]
LGTM :)

File: euphony/src/main/cpp/core/source/Base64.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            throw Base64Exception();
+    }
+}
+
+char Base64::convertInt2Char(int source) const {
+    const char base64Array[64] = {'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M',
+                                  'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z',
+                                  'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm',
+                                  'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
+                                  '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '+', '/'};
```
Comment (designe) [inline euphony/src/main/cpp/core/source/Base64.cpp:70]: Same with standard Base64 characters order! :)
image

**Review by designe — APPROVED** [review]
LGTM! :)
We already have a detail review in Base32.
https://github.com/euphony-io/euphony/pull/79
This base64 is also similar algorithm.

you did very well!
Thanks for your contribution!

**Comment by jopopcorn** [comment]
Your contribution is amazing and wonderful! Great job!

**Pull request merged**